javascript.md

```diff
@@ -132,8 +151,26 @@ jane.logHello(["John", "Sue"]);
 
 Rust:
 
+Note that Rust doesn't bind a context/scope like JavaScript does, so there isn't a need for a fat arrow syntax to opt out of the `this` binding.
+
 ```rust
-// TODO
+struct Person {
+    name: String,
+}
+
+impl Person {
+    fn log_hello(self, friends: Vec<&str>) -> String {
+        friends.iter().map(|friend| {
+            format!("{} says hello to {}", self.name, friend)
```

Comment:
Ok, so, each {} is substituted in format!, with format being a variatic function. 

Q: can one define different `format!` fucntions? Like in JS, you can do:
```JS
function  HTML(){ /* ... process this html template string ...*/}
\\template string
HTML`${value}`;  \\ call the HTML function when processing this template string.